src/java/org/apache/cassandra/schema/TableId.java

```diff
@@ -58,8 +53,7 @@ public final class TableId implements Comparable<TableId>
 {
     public static final long MAGIC = 1956074401491665062L;
     public static final long EMPTY_SIZE = ObjectSizes.measureDeep(new UUID(0, 0));
-
-    private static final ConcurrentHashMap<TableId, TableId> internCache = new ConcurrentHashMap<>();
```

Comment:
spoke with benedict about this, because we clean up this cache we don't have the property that `tableA == tableB`, so you *still* need to do `tableA.equals(tableB)`

He also has a different patch pending that can own this logic